Profile page game grid additions

BenBrownCreative opened the pull request:
Adds game details to a users games grid when a game is clicked on. Game details data from board game geek averages and from the users plays are combined to display nicely formatted stats to the user. UI improvements for the FocusedGame component. 

![image](https://user-images.githubusercontent.com/33357404/91105545-eb168c00-e635-11ea-8cb6-84a9e56b00ab.png)

meyer-mcmains commented on client/src/pages/Profile/index.js:
```diff
@@ -37,6 +38,7 @@ const Profile = props => {
       username: USERNAME
     }
   });
+  console.log(data);
```
```suggestion
```

BenBrownCreative replied:
removed

meyer-mcmains commented on client/src/components/PlayersLastPlayed/index.js:
```diff
@@ -0,0 +1,36 @@
+  justify-self: center;
+`;
+
+const Player = ({ name, score, win, ...props }) => {
```
You don't need to spread over props on every component, usually this is done if you are going to pass props down to another component
```suggestion
const Player = ({ name, score, win }) => {
```

BenBrownCreative replied:
good to know

meyer-mcmains commented on client/src/components/GameImage/index.js:
```diff
@@ -74,7 +83,8 @@ GameImage.propTypes = {
   transitionSpeed: PropTypes.string,
-  selected: PropTypes.bool
+  selected: PropTypes.bool,
+  username: PropTypes.string
```
I think this should be required?
```suggestion
  username: PropTypes.string.isRequired
```

BenBrownCreative commented:
Pushed up a refactored GameFocused file. I added an Icons component to handle the icon and text but was having trouble getting the formatting right to send the text prop as a template string and instead opted to send the text prop as an object. I am also sending the PlayerCount component as a prop to Icons as well.

I also added components.js files to GameFocused and Icons to separate styling from the main component. Perhaps PlayerCount and PlayTime should be moved somewhere else. Could not decide on the best place for them to live.

BenBrownCreative commented:
i implemented your changes and divided icons and iconWithText into separate components

meyer-mcmains commented on client/src/components/PlayersLastPlayed/index.js:
```diff
@@ -0,0 +1,32 @@
+const Player = ({ name, score, win }) => {
+  return (
+    <PlayerWrapper>
+      <IconWrapper>{win && <Trophy />}</IconWrapper>
```
I think this should use the new Icon component you made?

meyer-mcmains commented on client/src/components/GameFocused/components.js:
```diff
@@ -0,0 +1,96 @@
+`;
+
+export const ImgWrapper = styled.div`
+  /* background: ${p => p.theme.color.darkBlueHighlight}; */
```
should we remove this comment?
